File: Ejercicio 8/lot.java
abstract class Sensor {
    public abstract String leer();
    public abstract String reportar();
}

class SensorTemperatura extends Sensor {
    public String leer() { return "25°C"; }
    public String reportar() { return "Temperatura calida"; }
}

class SensorMovimiento extends Sensor {
    public String leer() { return "Movimiento humano detectado"; }
    public String reportar() { return "Alerta enviada al Banco"; }
}

public class Main {
    public static void main(String[] args) {
        Sensor[] sensores = {
            new SensorTemperatura(),
            new SensorMovimiento()
        };

        for (Sensor s : sensores) {
            System.out.println("- " + s.leer() + ", "  + s.reportar());
            
        }
    }
}

File: Ejercicio 8/lot.py
from abc import ABC, abstractmethod

class Sensor(ABC):

    @abstractmethod
    def leer(self):
        pass

    @abstractmethod
    def reportar(self):
        pass


class SensorTemperatura(Sensor):
    def leer(self):
        return "25°C"

    def reportar(self):
        return "Temperatura calida"


class SensorMovimiento(Sensor):
    def leer(self):
        return "Movimiento humano detectado"

    def reportar(self):
        return "Alerta enviada al Banco"



sensores = [SensorTemperatura(), SensorMovimiento()]

for s in sensores:
    print("-"+ s.leer() + ", "+ s.reportar())
